lib/fix/delete_asset_resources.rb

```diff
@@ -3,20 +3,68 @@
 
 module Fix
   class DeleteAssetResources < BatchProcess
+
     def run
-      asset_resources.each do |ar|
-        log.info "Destroying Asset Resource #{ar.id}"
-        begin
-          Hyrax.persister.delete(resource: ar)
-          Hyrax.index_adapter.delete(resource: ar)
-          Hyrax.index_adapter.connection.commit
-          log.info "Asset Resource #{ar.id} destroyed."
-        rescue => e
-          log_error e
-        end
+      log.info "Deleting #{asset_resources.count} Asset Resources..."
+      asset_resources.map do |ar|
+        destroy_work_and_members(resource: ar)
+      end
+      log.info "Done."
+      log.info report
+    end
+
+    private
+
+
+    def results; @results ||= []; end
+
+    def work_destroy_transaction
+      Hyrax::Transactions::WorkDestroy.new
+    end
+
+    def destroy_work_and_members(resource:)
+      result = { resource: resource }
+      resource.members.each do |member|
+        resource.members.delete(member)
```

Comment:
actually, since this is doing a depth-first recursive call to traverse the tree, removing that line really has no effect -- the everything gets deleted, children first. Removing.